common/src/main/java/io/netty/util/HashedWheelTimer.java

```diff
@@ -544,9 +545,9 @@ private long waitForNextTick() {
 
             for (;;) {
                 final long currentTime = System.nanoTime() - startTime;
-                long sleepTimeMs = (deadline - currentTime + 999999) / 1000000;
+                long sleepTimeNanos = deadline - currentTime + 999999;
```

Comment:
Can you explain what the +999999 here is for?  It seems to contradict the commit message, which offers sub millisecond ticks.

Reply:
@carl-mastrangelo yes you are right... this needs to be fixed.

Reply:
Could we sleep for ms and may follow another park in ns?